### ISSUE TITLE
A client can block the router (thread blocked in SocketChannel.send())

### ISSUE THREAD

**Issue by activeeon-bot** [ow2-proactive/programming#1006]
<a href="https://jira.activeeon.com/browse/PROACTIVE-1130" title="PROACTIVE-1130">Original issue</a> created by <a href="mailto:clement.mathieu_AT_activeeon.com">Clement Mathieu</a> on 19, Sep 2011 at 11:28 AM - PROACTIVE-1130

<hr />

The PAMR use a thread pool to forward messages to the clients. To send a message, the thread must lock the client to avoid data corruption. Each message is sent sequentially (TCP streams can't be shared). 

The issue is that if a SocketChannel.send() call block, then all the other messages for this client will wait on the lock and block a worker thread. Since the worker thread pool is bounded it could block the whole router for a long time (several hours on PACA grid).

We should find a way to avoid a such deadlock.

**Comment by activeeon-bot** [comment]
<a href="https://jira.activeeon.com/browse/PROACTIVE-1130?focusedCommentId=18627">Original comment</a> posted by <a href="mailto:youri.bonnaffe_AT_activeeon.com">Youri Bonnaffe</a> on 12, Dec 2014 at 13:07 PM

<hr />

Marking old issues as won't fix, either because we don't support it anymore, or it is needed or it has been there for too long.

Feel free to reopen if needed.
